pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -106,6 +97,46 @@ func (r *ReconcileNSTemplateSet) Reconcile(request reconcile.Request) (reconcile
 		reqLogger.Error(err, "failed to get NSTemplateSet")
 		return reconcile.Result{}, err
 	}
+	if util.IsBeingDeleted(nsTmplSet) {
+		// if the NSTemplateSet has no finalizer, then we don't have anything to do
+		if len(nsTmplSet.Finalizers) == 0 {
```

Comment:
```suggestion
		if !util.HasFinalizer(nsTmplSet, toolchainv1alpha1.FinalizerName) {
```

Reply:
will incorporate this change during the refactoring (moving the deletion part in a separate func)

Reply:
done in b1f9439